src/coreclr/jit/llvm.cpp

```diff
@@ -2698,32 +2774,75 @@ void Llvm::lowerToShadowStack()
     }
 }
 
+
 //------------------------------------------------------------------------
 // Convert GT_STORE_LCL_VAR and GT_LCL_VAR to use the shadow stack when the local needs to be GC tracked,
 // rewrite calls that returns GC types to do so via a store to a passed in address on the shadow stack.
 // Likewise, store the returned value there if required.
 //
-void Llvm::PlaceAndConvertShadowStackLocals()
+void Llvm::Lower()
 {
     populateLlvmArgNums();
 
     _shadowStackLocalsSize = 0;
 
     std::vector<LclVarDsc*> locals;
     unsigned localsParamCount = 0;
+    unsigned lvaCount = _compiler->lvaCount;
 
-    for (unsigned lclNum = 0; lclNum < _compiler->lvaCount; lclNum++)
+    for (unsigned lclNum = 0; lclNum < lvaCount; lclNum++)
     {
         LclVarDsc* varDsc = _compiler->lvaGetDesc(lclNum);
         if (!canStoreLocalOnLlvmStack(varDsc))
         {
             locals.push_back(varDsc);
             if (varDsc->lvIsParam)
             {
+                if (varDsc->lvPromotedStruct())
```

Comment:
```suggestion
                if (varDsc->lvPromoted)
```
`lvPromoted` means the same thing as `lvPromotedStruct` for us and the latter has been deleted in upstream.